targets/csharp/source/source/OneDSEventsAPI.cs

```diff
@@ -0,0 +1,182 @@
+
+using OneDSCsEvent = Microsoft.Applications.Events.DataModels.CsEvent;
+using OneDSEventData = Microsoft.Applications.Events.DataModels.Data;
+
+namespace PlayFab
+{
+    /// <summary>
+    /// Main interface for OneDS (One Data Collector) events APIs.
+    /// Write custom events to One Data Collector.
+    /// </summary>
+    public class OneDSEventsAPI
+    {
+        private string oneDSProjectIdIkey;
+        private string oneDSIngestionKey;
+
+        // "Tickets" are a special HTTP header in a POST request to OneDS server, e.g.:
+        // Tickets: "<ticket1_key>": "<ticket1_prefix>:<ticket1_value>";"<ticket2_key>": "<ticket2_prefix>:<ticket2_value>";...
+        // JWT token is provided as one of the tickets, e.g.:
+        // Tickets: "<oneDSHeaderJwtTicketKey>": "<oneDSHeaderJwtTicketPrefix>:<oneDSJwtToken>"; ...
+        private string oneDSJwtToken;
+        private string oneDSHeaderJwtTicketKey;
+        private string oneDSHeaderJwtTicketPrefix;
+
+        public bool IsOneDSAuthenticated { get; private set; }
+
+        public void SetCredentials(string projectIdIkey, string ingestionKey, string jwtToken, string headerJwtTicketKey, string headerJwtTicketPrefix)
+        {
+            this.oneDSProjectIdIkey = projectIdIkey;
+            this.oneDSIngestionKey = ingestionKey;
+            this.oneDSJwtToken = jwtToken;
+            this.oneDSHeaderJwtTicketKey = headerJwtTicketKey;
+            this.oneDSHeaderJwtTicketPrefix = headerJwtTicketPrefix;
+            this.IsOneDSAuthenticated = true;
+        }
+
+        public void ForgetAllCredentials()
+        {
+            this.IsOneDSAuthenticated = false;
+            this.oneDSProjectIdIkey = string.Empty;
+            this.oneDSIngestionKey = string.Empty;
+            this.oneDSJwtToken = string.Empty;
+            this.oneDSHeaderJwtTicketKey = string.Empty;
+            this.oneDSHeaderJwtTicketPrefix = string.Empty;
+        }
+
+        /// <summary>
+        /// Write batches of custom events to OneDS (One Data Collector).
+        /// The payload of custom events must be of OneDSEventData type.
+        /// </summary>
+        public async Task<PlayFabResult<WriteEventsResponse>> WriteTelemetryEventsAsync(WriteEventsRequest request, object customData = null, Dictionary<string, string> extraHeaders = null)
+        {
+            if (!request.Events.Any())
+            {
+                var apiMethodResult = new PlayFabResult<WriteEventsResponse>
+                {
+                    Error = new PlayFabError
+                    {
+                        Error = PlayFabErrorCode.ContentNotFound,
+                        ErrorMessage = "No events found in request. Please make sure to provide at least one event in the batch.",
+                        HttpStatus = "OneDSError"
+                    },
+                    CustomData = customData
+                };
+
+                PlayFabSettings.GlobalErrorHandler?.Invoke(apiMethodResult.Error);
```

Comment:
Mixed feelings about expanding the usage of this handler.
It is named "global" I guess...